python/ray/tests/test_actor.py

```diff
@@ -1676,5 +1677,59 @@ def method(self):
     assert result == "ok"
 
 
+def test_get_actor_after_same_name_actor_dead(shutdown_only):
+    ray.init(namespace="test")
+
+    @ray.remote
+    class Actor:
+        def get_pid(self):
+            return os.getpid()
+
+    def get_pid(actor):
+        while True:
+            try:
+                return ray.get(actor.get_pid.remote())
+            except Exception:
+                pass
+
+    a = Actor.options(name="test", max_restarts=1).remote()
+
+    pid = get_pid(a)
+    os.kill(pid, signal.SIGKILL)
+
+    pid = get_pid(a)
+    os.kill(pid, signal.SIGKILL)
+
+    wait_for_condition(lambda: ray.state.actors(a._actor_id.hex())["State"] == "DEAD")
+
+    # When a reference is held, the name cannot be reused.
+    with pytest.raises(ValueError):
+        Actor.options(name="test", max_restarts=1).remote()
+
+    # triger rpc reply: WaitForActorRefDeleted
```

Comment:
```suggestion
    # Deleting the remaining reference so the name can be reused
```
don't want to couple to comment to rpc name in case it changes